packages/jats/xref/XRefTargets.js

```diff
@@ -1,31 +1,71 @@
 'use strict';
 
 var Component = require('substance/ui/Component');
+var getXRefTargets = require('./getXRefTargets');
+var find = require('lodash/find');
+var without = require('lodash/without');
 
-/**
-  Takes a list of available reference targets (e.g. figure nodes)
-  and a list of ids (selected targets)
+/*
+  Editing of XRefTargets
 */
 function XRefTargets() {
   XRefTargets.super.apply(this, arguments);
 }
 
 XRefTargets.Prototype = function() {
 
+  this.getInitialState = function() {
+    return {
+      targets: getXRefTargets(this.props.node)
+    };
+  };
+
   this.render = function($$) {
     var el = $$('div').addClass('sc-xref-targets');
-    var node = this.props.node;
     var componentRegistry = this.context.componentRegistry;
-    var availableTargets = this.props.availableTargets;
 
-    availableTargets.forEach(function(targetNode) {
-      var TargetComponent = componentRegistry.get(targetNode.type+'-target');
+    this.state.targets.forEach(function(target) {
+      var TargetComponent = componentRegistry.get(target.node.type+'-target');
       el.append(
-        $$(TargetComponent, {node: targetNode})
+        $$(TargetComponent, target)
+          .on('click', this._toggleTarget.bind(this, target.node))
       );
-    });
+    }.bind(this));
     return el;
   };
+
+  this._toggleTarget = function(targetNode) {
+    var node = this.props.node;
+    var documentSession = this.context.documentSession;
+
+    // Update model
+    var newTargets = node.targets;
+    if (newTargets.indexOf(targetNode.id) > 0) {
+      newTargets = without(newTargets, targetNode.id);
+    } else {
+      newTargets.push(targetNode.id);
+    }
+
+    // TODO: this is causing the modal to close. Can we prevent that?
```

Comment:
@oliver---- you have an idea why the modal closes here? Can't figure out what's going on..